docs/source/execute_api.rst

```diff
@@ -0,0 +1,143 @@
+Executing notebooks
+===================
```

Comment:
The notebook exporter only really exists to fit execution+saving into the framework of nbconvert. It's not really necessary, and I think it's unwanted complexity for this example. Even if it can result in slightly fewer lines of code, it obscures what's going on.

 ExecutePreprocessor already handles getting the kernel name from the notebook metadata, so Exporter is no help there. Reading a notebook file with nbformat is a) trivial, and b) definitely a public API, so it's good to show people how to use it.

I think preprocessor order has been problematic precisely because nbconvert separates enabling the preprocessors from their priority. I think using them explicitly as Python objects rather than enabling them through config makes it easier to get them organised correctly.